### PR TITLE
renamed trial_space to be vector_space

### LINKED ISSUE #116: Rename TrialSpace as VectorSpace and create a derived TrialSpace concept.
Right now, we represent a linear subspace with a "TrialSpace". As we move to support more hyper-reduction techniques, we need to construct linear subspaces and it would be good to re-use what's in TrialSpace. We will handle this by re-naming TrialSpace as VectorSpace. Downstream we can then create the notion of a TrialSpace which has state-specific functions, i.e., mapFromReducedState, mapToReducedState, etc.

### PR DESCRIPTION
**Pull request by eparish1**
What to you all think of this?

### CODE REVIEW

File: romtools/vector_space/__init__.py [hunk 1]
```diff
@@ -61,13 +61,13 @@
 $\\mathbf{u}_{\\mathrm{shift}} \\in \\mathbb{R}^{ N_{\\mathrm{vars}}  N_{\\mathrm{x}}}$ is the shift vector (or affine offset),
 and, by definition, $\\mathcal{V} \\equiv \\mathrm{range}(\\boldsymbol \\Phi)$.
 
-The `TrialSpace` abstract class defined below encapsulates the information of an affine trial space, $\\mathcal{V}$,
-by virtue of providing access to a basis matrix, a shift vector, and the dimensionality of the trial space,
+The `VectorSpace` abstract class defined below encapsulates the information of an affine vector space, $\\mathcal{V}$,
+by virtue of providing access to a basis matrix, a shift vector, and the dimensionality of the vector space,
```
Comment (jtencer) [inline romtools/vector_space/__init__.py:65]: Do we want to keep "basis **matrix**" and "shift **vector**" here?

**Review by jtencer — APPROVED** [review]

File: romtools/vector_space/__init__.py (hunk 1 above)
Comment (fnrizzi) [reply romtools/vector_space/__init__.py:65]: i think we shuld do in a separate PR

Comment (eparish1) [reply romtools/vector_space/__init__.py:65]: A separate PR is fine by me for this.

**Comment by fnrizzi** [comment]
are you sure you removed all instances of trial_space? or was this PR just for the class?

**Comment by fnrizzi** [comment]
just grepped, i see:
```
./hyper_reduction/ecsw.py:trial basis.
./hyper_reduction/ecsw.py:weights given a set of residual snapshot and trial basis data.
./vector_space/__init__.py:In a ROM, a high-dimensional state is restricted to live within a low-dimensional vector space, known as a trial space.
./vector_space/__init__.py:- $\\mathcal{V}$ with $\\text{dim}(\\mathcal{V}) = K \\le N_{\\mathrm{vars}}  N_{\\mathrm{x}}$ is the trial space
./workflows/greedy/greedy_coupler_base.py:    def create_trial_space(self, training_sample_indices):
./workflows/greedy/greedy_coupler_base.py:        trial space.
./workflows/greedy/run_greedy.py: 5. We employ the first two solutions to compute the trial space, e.g.,
./workflows/greedy/run_greedy.py:    greedy_coupler.create_trial_space(training_dirs)
./workflows/greedy/run_greedy.py:        greedy_coupler.create_trial_space(training_dirs)
```

**Comment by eparish1** [comment]
@fnrizzi 

- The greedy coupler still uses the concept of create_trial_space, but this is intentional since it's specific to the ROM workflow.
- The documentation on vector space still has some discussion of ROM trial space; I was thinking to keep this until (if) we add an additional trial space layer. I'm glad to revisit this.
- For ecsw, this is not related to the vector space stuff as far as implementation goes.

**Comment by fnrizzi** [comment]
yes just making sure, makes sense!

**Pull request merged**